app/components/app_patient_programmes_table_component.rb

```diff
@@ -25,63 +25,143 @@ def call
   HEADERS = ["Programme name", "Status", "Notes"].freeze
 
   def rows
-    programmes.flat_map { |programme| rows_for_programme(programme) }
+    programmes.flat_map { |programme| rows_for_programme(programme:) }
   end
 
-  def rows_for_programme(programme)
+  def rows_for_programme(programme:)
     if programme.seasonal?
-      seasonal_programme_rows(programme)
+      seasonal_programme_rows(programme:)
     else
-      non_seasonal_programme_rows(programme)
+      non_seasonal_programme_rows(programme:)
     end
   end
 
-  def seasonal_programme_rows(programme)
+  def seasonal_programme_rows(programme:)
     eligible_year_groups = eligible_year_groups_for(programme:)
 
     AcademicYear.all.flat_map do |academic_year|
       year_group = patient.year_group(academic_year:)
       next unless year_group.in?(eligible_year_groups)
 
-      if vaccinated_for_academic_year?(programme:, academic_year:)
-        vaccination_records_for(programme).map do |vaccination_record|
-          build_row(programme, academic_year, vaccination_record)
+      vaccination_records = vaccination_records_for(programme:, academic_year:)
+
+      rows =
+        (1..vaccination_records.count - 1).to_a.map do |dose_sequence|
```

Comment:
I'm not sure this will work as intended:

- I'm not seeing anything in `vaccination_records_for` that will order the records by dose sequence, meaning the `dose_sequence` here may not line up with the `dose_sequence` on the record.
- Due to poor historical data some patients will have duplicate vaccination records, with the same `dose_sequence` value.
- Some vaccination records (for MenACWY and Td/IPV) [won't have a `dose_sequence`](https://github.com/nhsuk/manage-vaccinations-in-schools/blob/14250709d78c37b1ffbf12516a290cab18c4c87e/app/models/programme.rb#L97). This is because we may not have a complete picture of the patients vaccination history at the time of recording and therefore can't set one.